### ISSUE TITLE
Any change of getting integrity check?

### ISSUE THREAD

**Issue by dougg0k** [Pr0methean/zip#38]
Hey,

Would you be able to add a integrity check to test each file?

eg.,
https://unix.stackexchange.com/questions/197127/test-integrity-of-zip-file

**Comment by im7mortal** [comment]
After analyzing the request, I’ve come to the conclusion that this feature does not really belong in the library itself.

The reporter references a Stack Overflow answer suggesting `zip -t` and `zip -T`. However, this comparison mixes two different layers:

- `zip` is a finished CLI program
- `zip2` is a library

Because `zip2` is a library, it should expose building blocks rather than replicate CLI-style flags. If a `zip2` user needs behavior equivalent to `-t` or `-T`, it can be easily implemented in user code using the existing API.

## 1. Equivalent of zip -t

`zip -t` checks the integrity of an existing ZIP file.
This is straightforward to implement with `zip2` by reading all entries and fully consuming their contents (CRC verification happens during read).

Example:
```rust
use std::error::Error;
use std::fs::File;
use std::io::{self};
use zip::ZipArchive;

fn main() -> Result<(), Box<dyn Error>> {
    let zip_file = File::open("/home/user/Downloads/Apidog-linux-deb-latest.zip")?;
    let mut archive = ZipArchive::new(zip_file)?;

    for i in 0..archive.len() {
        let mut file = archive.by_index(i)?;
        let mut sink = io::sink();
        io::copy(&mut file, &mut sink)?;
    }

    Ok(())
}
```

If this function completes without error, the ZIP file is valid.

## 2. Equivalent of zip -T

`zip -T` simply tests the archive after it has been created.
With `zip2`, this is just a matter of calling the same validation logic once ZIP creation is finished.

In other words, `-T` is not a separate feature—it’s just `-t` applied immediately after writing.

**Comment by Its-Just-Nans** [comment]
Linked with https://github.com/zip-rs/zip2/pull/235

**Comment by Its-Just-Nans** [comment]
> If this function completes without error, the ZIP file is valid.

The ZIP file may be valid but it can also contains slight inaccuracies. Since the zip reading allow it
